modules/api/docs/LorisRESTAPI_v0.0.4-dev.md

```diff
@@ -823,90 +823,192 @@ The JSON object is of the form:
 
 ```js
 {
     "Meta" : {
         "CandID" : $CandID,
         "Visit" : $VisitLabel,
     },
-    "DicomTars" : 
-    [
+    "DicomTars" : [
         {
             "Tarname" : "DCM_yyyy-mm-dd_ImagingUpload-hh-mm-abc123.tar",
             "Patientname" : "DCM123_123456_V1",
-            "SeriesInfo" :
-                [{
+            "SeriesInfo" : [
+                {
                     "SeriesDescription" : "MPRAGE_ipat2",
-                    "SeriesNumber" : "2",
+                    "SeriesNumber" : 2,
                     "EchoTime" : "2.98",
                     "RepetitionTime" : "2300",
                     "InversionTime" : "900",
                     "SliceThickness" : "1",
                     "Modality" : "MR",
                     "SeriesUID" : "1.2.3.4.1107",
-                    },
-                    {
+                },
+                {
                     "SeriesDescription" : "BOLD Resting State",
-                    "SeriesNumber" : "5",
+                    "SeriesNumber" : 5,
                     "EchoTime" : "30",
                     "RepetitionTime" : "2100",
                     "InversionTime" : NULL,
                     "SliceThickness" : "3.5",
                     "Modality" : "MR",
                     "SeriesUID" : "3.4.5.6.1507",
-                }]
+                }
+            ]
         },
         {
             "Tarname" : "DCM_yyyy-mm-dd_ImagingUpload-hh-mm-def456.tar",
             "Patientname" : "DCM456_654321_V1",
-            "SeriesInfo" :
-                [{
-                "SeriesDescription" : "MPRAGE_ipat2",
-                "SeriesNumber" : "2",
-                "EchoTime" : "2.98",
-                "RepetitionTime" : "2300",
-                "InversionTime" : "900",
-                "SliceThickness" : "1",
-                "Modality" : "MR",
-                "SeriesUID" : "1.7.8.9.1296",
-                }]
+            "SeriesInfo" : [
+                {
+                  "SeriesDescription" : "MPRAGE_ipat2",
+                  "SeriesNumber" : 2,
+                  "EchoTime" : "2.98",
+                  "RepetitionTime" : "2300",
+                  "InversionTime" : "900",
+                  "SliceThickness" : "1",
+                  "Modality" : "MR",
+                  "SeriesUID" : "1.7.8.9.1296",
+                }
+            ]
         }
-    ]    
+    ]
 }
 ```
 
-The `Modality` header in the SeriesInfo is either `MR` or `PT` for MRI or PET 
+The `Modality` header in the SeriesInfo is either `MR` or `PT` for MRI or PET
 scans, respectively.
 
+```
+POST /candidates/$CandID/$VisitLabel/dicoms
+```
+
+The body of the POST request should be encoded as a form data and contain a
+field `File` with the imaging file, and `Json`, with the following keys and
+values:
+
+```js
+{
+    "CandID": $CandID,
+    "PSCID": $PSCID,
+    "VisitLabel": $VisitLabel,
+    "IsPhantom": boolean
+}
+```
+
+An optionnal header 'LORIS-Overwrite' with the value 'overwrite' can be used to
```

Comment:
The `LORIS-Overwrite` feature does not work. I still get the following error if the file is already in the incoming directory:
```
<Response [409]>
{'error': 'This file has already been uploaded.'}
```

Also, wouldn't it be better to set `LORIS-Overwrite` to a boolean instead: `true` if want to overwrite existing file, `false otherwise`? And maybe call it `OverwriteFileIfAlreadyExistOnServer` or something like that to make it more explicit what this option will do if set to True.

Reply:
Hhhhm, the setting seemed to work on my machine, I'd be curious to know how you passed the argument (if you tested using my code or yours for instance). I think `OverwriteFileIfAlreadyExistOnServer` is a bit long and should rather belong in the documentation, but no strong opinion, either way is fine with me.

Reply:
yes, I agree, it is definitely too long. Maybe `Upload-Overwrite` would make more sense than `LORIS-Overwrite` in this case though. Just to make it clearer?

Reply:
Below is the code I am using (after having uploading the scan once):
```
dicom_upload = {
    'CandID': '300001',
    'PSCID': 'MTL001',
    'VisitLabel': 'V2',
    'IsPhantom': False,
    'LORIS-Overwrite': 'overwrite'
}
file_path = '/Users/cmadjar/Data/LORIS_imaging_data/incoming/test_api_upload/MTL001_300001_V2.tgz'
url = os.path.join(api_cred['host'], 'api', api_cred['version'], 'candidates', '300001', 'V2', 'dicoms')
try:
    response = requests.post(
        url=url,
        headers={'Authorization': 'Bearer %s' % jwtoken},
        data={'Json': json.dumps(dicom_upload)},
        allow_redirects=False,
        files={'File': (os.path.basename(file_path), open(file_path, 'rb'), 'application/tar')}

    )
    print(response)
    print(response.json())
```
When overwriting the file, I get the following response:
```
<Response [409]>
{'error': 'This file has already been uploaded.'}
```

So I m not able to overwrite the file. Not sure why it works for you but not me.

Reply:
Oh, it is because you are passing `LORIS-Overwrite: overwrite` as a JSON attribute, whereas it should be passed as an HTTP header. Try changing these lines:
```diff
-        headers={'Authorization': 'Bearer %s' % jwtoken},
+        headers={
+            'Authorization': 'Bearer %s' % jwtoken,
+            'LORIS-Overwrite': 'overwrite',
+        }
```
The parameter passed in the HTTP headers was in the original PR, I don't know what is the rationale here, so no opinion on HTTP header vs JSON attribute on my part.

Reply:
ah ben oui. Haha. OK. Not sure if the doc should be made clearer somehow? I guess it says 'optional header' but I did not understand that it was referring to the HTTP headers. Maybe it was just not clear for me and other developers would get it.

Reply:
BTW, when you call it correctly, it works ;) Glad I put the code snippet in my comment!